_practicalities/intro.md

```diff
@@ -0,0 +1,355 @@
+disruption. Please still check that the libraries you use follow best practices
+not to break for Python 2 users. Python is a community regardless of which
+python version you have to (or decided to) run, making sure that everything
+works make the community strong.
+
+Make sure you have Pip >= 9.0, this is especially important if you have Python
+2 installations. Having pip 9.0+ is not a guaranty to flawless upgrade. But pip
+9.0+ does have a number of safety check not available on previous versions.
+
+Having a version of pip < 9.0 can lead your system to try to upgrade to
+non-compatible versions of Python packages even if these are marked as
+non-compatible.
+
+Help as many other _users_ as possible to install pip >=9.0, for the
+transition, it is the slowest part of the ecosystem to update, and is the only
+piece that requires action of all Python users.
+
+The simplest way to make sure all is up to date is to run the following for
+each installation of Python:
+
+    $ pip install --upgrade setuptools pip
+
+This will install the latest version of pip and setuptools.
+
+You can issue the following to see the version of pip:
+
+   $ pip --version
+   9.0.0
+
+All good.
+
+
+
+## Setuptools
+
+If you are on a system  for which no wheel is available, pip will try to
+install a source distribution (aka `sdist`).
+
+Installing an `sdist` will require setuptools make sure you have setuptools
+`>=24.2.0` (mnemonic: 2-42, [why
+42](https://en.wikipedia.org/wiki/The_answer_to_life_the_universe_and_everything)
+?) or building Python 3 only libraries is likely to fail. In particular if
+library authors have taken time to mark their library as Python 3 only, the
+`python_requires` argument to `setup()` may not be recognized and installation
+will fail.
+
+Use the following to check setuptools version :
+
+    $ python -c 'import setuptools; print(setuptools.__version__)
+    24.2.0
+
+Again make sure to upgrade pip and setuptools to make sure you have an up to
+date system:
+
+    $ pip install --upgrade setuptools pip
+
+## Local package index
+
+If you are using a custom local package index, for example if you are working
+at a company with private packages, make sure it implement correctly
+[pep-503](https://www.python.org/dev/peps/pep-0503/) and let pip knows about
+the `python_requires` field. This _mostly_ mean that the html you are exposing
+should get a `data-python-requires` data attribute with the (html escaped)
+version specifier.
+
+## The state of PyPI
+
+Note that at the time of this writing the patches to `pypi.python.org` are not
+deployed yet but should hopefully be deployed soon.
+
+[Warehouse](https://github.com/pypi/warehouse) and [Legacy
+PyPI](https://github.com/pypa/legacy-pypi) have received various patches to
+insure they support this new functionality.
+
+You can publish a package with the `requires_python` metadata **now**, it will
+be correctly exposed once pypi is deployed.
```

Comment:
This appears to be working now, no need to wait for PyPI to be deployed again. https://pypi.python.org/simple/flit/